hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java

```diff
@@ -469,15 +478,76 @@ public boolean delete(Path f, boolean recursive) throws IOException {
         return false;
       }
 
+      OFSPath ofsPath = new OFSPath(key);
+
+      // Handle rm root
+      if (ofsPath.isRoot()) {
+        Iterator<? extends OzoneVolume> it =
+            adapterImpl.getObjectStore().listVolumes("");
+        while (it.hasNext()) {
+          OzoneVolume volume = it.next();
+          String nextVolume = addTrailingSlashIfNeeded(
+              f.toString()) + volume.getName();
+          delete(new Path(nextVolume), true);
+        }
```

Comment:
Replace the rm root logic with a warning. Effectively disallowing admin to use `rm -r /` to clear the cluster.

Admins should re-init the cluster if they really want to clear the cluster.